### PR TITLE
EZP-31103: Introduced strict types for ContentTypeService

### PR DESCRIPTION
**Pull request by adamwojs**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31103](https://jira.ez.no/browse/EZP-31103)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `master`
| **BC breaks**      | yes
| **Tests pass**     | yes
| **Doc needed**     | yes

This PR introduces strict types for the `\eZ\Publish\API\Repository\ContentTypeService`

**TODO**:
- [X] Implement feature / fix a bug.
- [X] Implement tests.
- [X] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/MVC/Symfony/Templating/Tests/Twig/Extension/ContentExtensionTest.php [hunk 1]
```diff
@@ -35,6 +35,8 @@ class ContentExtensionTest extends FileSystemTwigIntegrationTestCase
 
     private $fieldDefinitions = [];
 
+    private $identityMap = [];
```
Comment (mikadamczyk) [inline eZ/Publish/Core/MVC/Symfony/Templating/Tests/Twig/Extension/ContentExtensionTest.php]: We could add docblock here and above

File: eZ/Publish/Core/MVC/Symfony/Templating/Tests/Twig/Extension/ContentExtensionTest.php [hunk 2]
```diff
@@ -70,12 +72,18 @@ public function getFixturesDir()
      */
     protected function getContent($contentTypeIdentifier, array $fieldsData, array $namesData = [])
```
Comment (mikadamczyk) [inline eZ/Publish/Core/MVC/Symfony/Templating/Tests/Twig/Extension/ContentExtensionTest.php]:
```suggestion
    protected function getContent(string $contentTypeIdentifier, array $fieldsData, array $namesData = [])
```

File: eZ/Publish/Core/Persistence/Legacy/Content/Type/Gateway/DoctrineDatabase.php [hunk 3]
```diff
@@ -158,7 +158,7 @@ public function insertGroup(Group $group)
         );
         $q->prepare()->execute();
 
-        return $this->dbHandler->lastInsertId(
+        return (int)$this->dbHandler->lastInsertId(
```
Comment (mikadamczyk) [inline eZ/Publish/Core/Persistence/Legacy/Content/Type/Gateway/DoctrineDatabase.php]: If we add `(int)` here than we should change also `@return mixed Group ID`  https://github.com/ezsystems/ezpublish-kernel/pull/2856/files#diff-f7709a1560b54a55dbc2fd87a1ee5bd9R133

File: eZ/Publish/Core/Repository/ContentTypeService.php [hunk 4]
```diff
@@ -867,7 +865,7 @@ public function loadContentType($contentTypeId, array $prioritizedLanguages = []
     /**
      * {@inheritdoc}
      */
-    public function loadContentTypeByIdentifier($identifier, array $prioritizedLanguages = [])
+    public function loadContentTypeByIdentifier(string $identifier, array $prioritizedLanguages = []): ContentType
     {
         if (!is_string($identifier)) {
```
Comment (mikadamczyk) [inline eZ/Publish/Core/Repository/ContentTypeService.php]: This check is no longer needed

File: eZ/Publish/Core/Repository/ContentTypeService.php [hunk 5]
```diff
@@ -1517,7 +1515,7 @@ public function publishContentTypeDraft(APIContentTypeDraft $contentTypeDraft)
      *
      * @return \eZ\Publish\API\Repository\Values\ContentType\ContentTypeGroupCreateStruct
      */
-    public function newContentTypeGroupCreateStruct($identifier)
+    public function newContentTypeGroupCreateStruct(string $identifier): ContentTypeGroupCreateStruct
     {
         if (!is_string($identifier)) {
```
Comment (mikadamczyk) [inline eZ/Publish/Core/Repository/ContentTypeService.php]: This check is no longer needed

File: eZ/Publish/Core/Repository/ContentTypeService.php [hunk 6]
```diff
@@ -1539,7 +1537,7 @@ public function newContentTypeGroupCreateStruct($identifier)
      *
      * @return \eZ\Publish\API\Repository\Values\ContentType\ContentTypeCreateStruct
      */
-    public function newContentTypeCreateStruct($identifier)
+    public function newContentTypeCreateStruct(string $identifier): APIContentTypeCreateStruct
     {
         if (!is_string($identifier)) {
```
Comment (mikadamczyk) [inline eZ/Publish/Core/Repository/ContentTypeService.php]: This check is no longer needed

File: eZ/Publish/Core/Repository/ContentTypeService.php [hunk 7]
```diff
@@ -1583,7 +1581,7 @@ public function newContentTypeGroupUpdateStruct()
      *
      * @return \eZ\Publish\API\Repository\Values\ContentType\FieldDefinitionCreateStruct
      */
-    public function newFieldDefinitionCreateStruct($identifier, $fieldTypeIdentifier)
+    public function newFieldDefinitionCreateStruct(string $identifier, string $fieldTypeIdentifier): FieldDefinitionCreateStruct
     {
         if (!is_string($identifier)) {
```
Comment (mikadamczyk) [inline eZ/Publish/Core/Repository/ContentTypeService.php]: This check is no longer needed

File: eZ/Publish/SPI/Repository/Tests/Decorator/ContentTypeServiceDecoratorTest.php [hunk 8]
```diff
@@ -139,7 +139,7 @@ public function testLoadContentTypeDecorator()
         $decoratedService = $this->createDecorator($serviceMock);
 
         $parameters = [
-            'random_value_5ced05ce102297.95671709',
+            1,
             ['random_value_5ced05ce1022b3.56541050'],
```
Comment (mikadamczyk) [inline eZ/Publish/SPI/Repository/Tests/Decorator/ContentTypeServiceDecoratorTest.php]:
```suggestion
             ['prioritized_language_value'],
```

File: eZ/Publish/SPI/Repository/Tests/Decorator/ContentTypeServiceDecoratorTest.php [hunk 9]
```diff
@@ -55,7 +55,7 @@ public function testLoadContentTypeGroupDecorator()
         $decoratedService = $this->createDecorator($serviceMock);
 
         $parameters = [
-            'random_value_5ced05ce0ffda1.73499446',
+            1,
             ['random_value_5ced05ce0ffde5.37998562'],
```
Comment (mikadamczyk) [inline eZ/Publish/SPI/Repository/Tests/Decorator/ContentTypeServiceDecoratorTest.php]:
```suggestion
             ['prioritized_language_value'],
```

**Comment by adamwojs** [comment]
PR rebased and updated according to @mikadamczyk code review suggestions.

**Comment by adamwojs** [comment]
PR updated according to @mikadamczyk @ViniTou code review suggestions.

File: eZ/Publish/Core/Persistence/Legacy/Content/Type/Gateway/DoctrineDatabase.php [hunk 10]
```diff
@@ -130,7 +130,7 @@ public function __construct(DatabaseHandler $db, Connection $connection, MaskGen
      *
      * @param \eZ\Publish\SPI\Persistence\Content\Type\Group $group
      *
-     * @return mixed Group ID
+     * @return int Group ID
      */
     public function insertGroup(Group $group)
```
Comment (alongosz) [inline eZ/Publish/Core/Persistence/Legacy/Content/Type/Gateway/DoctrineDatabase.php]: Anything stopping us now from adding return type?
```suggestion
    public function insertGroup(Group $group): int
```

**Comment by micszo** [comment]
Testing discovered: https://jira.ez.no/browse/EZP-31131

**Comment by adamwojs** [comment]
FYI: The issue mentioned above has been fixed in https://github.com/ezsystems/ezplatform-admin-ui/pull/1133

**Comment by micszo** [comment]
Testing discovered: https://jira.ez.no/browse/EZEE-2930

**Pull request merged**